MatchGameTests/GameTests.cs

```diff
@@ -0,0 +1,136 @@
+        [InlineData(24)]
+        public void DeclareWinner_ReturnsDrawWhenPlayerScoresAreEqual(int matchingScore)
+        {
+            //Arrange
+            _player1Mock.Setup(x => x.Score).Returns(matchingScore);
+            _player2Mock.Setup(x => x.Score).Returns(matchingScore);
+
+            ResultEnum expected = ResultEnum.DRAW;
+            //Act
+            ResultEnum actual = _sut.DeclareWinner();
+            //Assert
+            Assert.Equal(expected, actual);
+
+        }
+
+        [Fact]
+        public void DeclareWinner_ReturnsPlayer1WhenPlayer1ScoreIsGreater()
+        {
+            //Arrange
+            _player1Mock.Setup(x => x.Score).Returns(1);
+            _player2Mock.Setup(x => x.Score).Returns(0);
+
+            ResultEnum expected = ResultEnum.PLAYER1;
+            //Act
+            ResultEnum actual = _sut.DeclareWinner();
+            //Assert
+            Assert.Equal(expected, actual);
+
+        }
+
+        [Fact]
+        public void DeclareWinner_ReturnsPlayer2WhenPlayer1ScoreIsLess()
+        {
+            //Arrange
+            _player1Mock.Setup(x => x.Score).Returns(1);
+            _player2Mock.Setup(x => x.Score).Returns(23);
+
+            ResultEnum expected = ResultEnum.PLAYER2;
+            //Act
+            ResultEnum actual = _sut.DeclareWinner();
+            //Assert
+            Assert.Equal(expected, actual);
+
+        }
+
+        [Fact]
+        public void Setup_SetsTheMatchingRuleForGamePlay()
+        {
+            //Arrange
+            MatchRuleEnum matchRule = MatchRuleEnum.FULL;
+            //Act
+            var _ = _sut.Setup(matchRule, 1);
+            MatchRuleEnum actual = _sut.MatchRule;
+            //Assert
+            Assert.Equal(matchRule, actual);
+        }
+
+        [Fact]
+        public void Setup_InformsWhenFullMatchingRuleIsIncompatibleWithASinglePackDeck()
+        {
+            //Arrange
+            MatchRuleEnum matchRule = MatchRuleEnum.FULL;
+            int singlePack = 1;
+            string expected = "You can play with this setup, but nobody ever wins!";
+            
+            //Act
+            string actual= _sut.Setup(matchRule, singlePack);
+
+            //Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void Setup_InformsThatGameIsReadyToPlay()
+        {
+            //Arrange
+            MatchRuleEnum matchRule = MatchRuleEnum.FULL;
+            string expected = "The game is ready to play.";
+
+            //Act
+            string actual = _sut.Setup(matchRule, 3);
+
+            //Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void Play_ReturnsFromGameLoopWhenDeckHasNoCardsRemaining()
+        {
+            //Arrange
+            _deckMock.Setup(x => x.HasCards).Returns(false);
+            //Act
+            bool gameOver = _sut.Play();
+            //Assert
+            Assert.True(gameOver);
+        }
+
```

Comment:
Game tests are really lacking. Some observable behaviours are: 
- IsMatch() is called `(numOfPacks * deckSize)-1` times.
- TakeCard() is called  `(numOfPacks * deckSize)` times.
- StartNewPile() is called at least once
- PlayToPileUntilMatch() is called at least once

With some considered test doubles created: 
- test that if one player recieves all the DelareMatch()'s in  sequence, they will be the winnner.